packages/edit-navigation/src/components/menu-editor/use-navigation-blocks.js

```diff
@@ -1,142 +1,184 @@
+					)
 				);
-				menuItemsRef.current[ block.clientId ] = item;
-				innerBlocks.push( block );
 			}
-			return innerBlocks;
-		};
-
-		// createMenuItemBlocks takes an array of top-level menu items and recursively creates all their innerBlocks
-		const innerBlocks = createMenuItemBlocks( itemsByParentID[ 0 ] );
-		setBlocks( [ createBlock( 'core/navigation', {}, innerBlocks ) ] );
-	}, [ menuItems ] );
-
-	const saveBlocks = () => {
-		const { clientId, innerBlocks } = blocks[ 0 ];
-		const parentItemId = menuItemsRef.current[ clientId ]?.parent;
-
-		const saveNestedBlocks = async ( nestedBlocks, parentId = 0 ) => {
-			for ( const block of nestedBlocks ) {
-				const menuItem = menuItemsRef.current[ block.clientId ];
-				let currentItemId = menuItem?.id || 0;
-
-				if ( ! menuItem ) {
-					const savedItem = await saveMenuItem( {
-						...createMenuItemAttributesFromBlock( block ),
-						menus: menuId,
-						parent: parentId,
-					} );
-					if ( block.innerBlocks.length ) {
-						currentItemId = savedItem.id;
-					}
-				}
+		},
+		[ debouncedBlocks ]
+	);
 
-				if (
-					menuItem &&
-					! isEqual(
-						block.attributes,
-						createBlockFromMenuItem( menuItem ).attributes
-					)
-				) {
-					saveMenuItem( {
-						...menuItem,
-						...createMenuItemAttributesFromBlock( block ),
-						menus: menuId, // Gotta do this because REST API doesn't like receiving an array here. Maybe a bug in the REST API?
-						parent: parentId,
-					} );
-				}
+	// Save handler
+	const { receiveEntityRecords } = useDispatch( 'core' );
+	const { createSuccessNotice, createErrorNotice } = useDispatch(
+		'core/notices'
+	);
 
-				if ( block.innerBlocks.length ) {
-					saveNestedBlocks( block.innerBlocks, currentItemId );
-				}
-			}
-		};
+	const saveBlocks = async () => {
+		const result = await batchSave( menuId, menuItemsRef, blocks[ 0 ] );
+
+		if ( result.success ) {
+			const mapping = menuItemsRef.current;
+			const allMenuItemsIds = Object.values( menuItemsRef.current ).map(
+				( { id } ) => id
+			);
+			const savedMenuItemIds = getAllClientIds( blocks ).map(
+				( clientId ) => mapping[ clientId ].id
+			);
+			setDeletedMenuItemsIds( [
+				...deletedMenuItemsIds,
+				difference( allMenuItemsIds, savedMenuItemIds ),
+			] );
+
+			createSuccessNotice( __( 'Navigation saved.' ), {
+				type: 'snackbar',
+			} );
+			receiveEntityRecords( 'root', 'menuItem', [], query, true );
+		} else {
+			createErrorNotice( __( 'There was an error.' ), {
+				type: 'snackbar',
+			} );
+		}
+	};
 
-		saveNestedBlocks( innerBlocks, parentItemId );
+	return [
+		blocks,
+		setBlocks,
+		() => promiseQueueRef.current.then( saveBlocks ),
+	];
+}
 
-		const deletedClientIds = difference(
-			Object.keys( menuItemsRef.current ),
-			innerBlocks.map( ( block ) => block.clientId )
-		);
+async function createDraftMenuItem() {
+	return apiFetch( {
+		path: `/__experimental/menu-items`,
+		method: 'POST',
+		data: {
+			title: 'Placeholder',
```

Comment:
Can't we create the draft with the values it has instead of Placeholder?

Reply:
It doesn't have any values - this is called immediately after the navigation link block is created